tests/typecheck/test_shortcuts.yml

```diff
@@ -39,14 +39,14 @@
 
 -   case: check_render_function_arguments_annotations
     main: |
-        from typing import Any
-        from typing_extensions import TypedDict
+        from typing import Any, Union
         from django.shortcuts import render
+        from django.template import Node
         from django.http.request import HttpRequest
 
-        TestContext = TypedDict("TestContext", {"user": Any})
+        TestContext = dict[Union[int, str, Node], Any]
```

Comment:
I think that this test is here to be sure that custom `TypedDict` instances are allowed. Because people can type their contexts with it.

Please, do not change it.

Reply:
I understand the use case. However, I'm not sure on the way forward here. 

As far as I can tell [ `TypedDict`](https://peps.python.org/pep-0589/#abstract) only supports `the use case where a dictionary object has a specific set of string keys`. 

The change I'm proposing here is that the key could be `int | str | Node`, and therefore the two are incompatible?

One option is not to change the type in the shortcut file. This would allow the test to pass, but I'm not sure it's the right solution.

Reply:
I think that typed dict `{'user': anything}` must be compatible with `dict[int | str | None, Any]`. But, if it is not - I am not surprised. 

I've looked into the test history - it is quite old. I don't remember the history behind it. But, it is clear that we can break this use-case for user who already use it. *But*, using correct type annotations is also quite important.

I think people can go with `cast()` when using typed dicts, so it is more important to have correct annotations 🤔

Reply:
I found `TypedDict` not being compatible with `dict` somewhat surprising. However, that's been discussed previously at https://github.com/typeddjango/django-stubs/issues/355 and so maybe we need to use `Mapping` in place of `Dict` throughout? 

I'm also surprised that a typed dict is not compatible with a wider type definition. The same is true of `dict`'s too. 

``` python
from typing import TypedDict, Any, Mapping

class MyContext(TypedDict):
    user: Any


def str_mapping_function(context: Mapping[str, Any]) -> None:
    pass

def multi_mapping_function(context: Mapping[str | int , Any]) -> None:
    pass

ctx = MyContext(user="MyName")

str_mapping_function(ctx) # ok
multi_mapping_function(ctx) #  error: Argument 1 to "multi_mapping_function" has incompatible type "MyContext"; expected "Mapping[Union[str, int], Any]"
```

Reply:
Can you please try `Mapping`?

Reply:
I updated to mapping, and reverted the test. Just left with one error which I'm not sure how to resolve. 

`main:9: error: Argument 3 to "render" has incompatible type "TestContext"; expected "Optional[Mapping[Union[int, str, Node], Any]]" (diff)`

I could add a `cast()` like this, but this doesn't seem like a good answer to me.
``` python
test_context = cast(Mapping[Union[int, str, Node], Any], test_context)
```

Reply:
We can remove this test are recommend users to use `cast(Any, typed_context)` for their code. If they use `TypedDict`.